src/fm.c

```diff
@@ -8,12 +12,39 @@ int main(int argc, char const *argv[])
         fprintf(stderr,
                 "Usage: %s -p genome\n       %s genome reads\n",
                 argv[0], argv[0]);
+        fprintf(stderr,
+                "argc=%d\n",
+                argc);
         return 1;
     }
+    char* processFileName = get_file_name_by_fa(argv[2]);
+
     if (strcmp("-p", argv[1]) == 0)
     {
         // preprocessing
-        printf("Preprocessing genome %s\n", argv[2]);
+        //printf("Preprocessing genome %s\n", argv[2]);
+        FILE* processFile = get_file(processFileName);
+        char* fasta_str = read_file(argv[2]);
+        char* fasta_str_start = fasta_str;
```

Comment:
Why is `fasta_str_start` necessary? I can see that this is the one you are `free()`'ing later (and it has to be the address you get from an allocation as the one you do in `read_file()`, but you don't update `fasta_str` anywhere, and you could thus just use that.